### DISCUSSION TITLE
Terser plugin Minimizer doesn't return result

### DISCUSSION THREAD

**Discussion by normancarcamo** [webpack/webpack#17877] [Q&A]
# Bug report

**What is the current behavior?**

Error when compiling for production using terser options

```bash
ERROR in ./javascripts/webpack/main.js from Terser plugin
Minimizer doesn't return result
```

```js
{
  optimization: {
    minimize: true,
    minimizer: [
      new TerserPlugin({
        minify: TerserPlugin.uglifyJsMinify,
        extractComments: false,
        terserOptions: {
          keep_classnames: true,
          keep_fnames: true,
        },
      }),
    ],
  },
}
```

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

It should compile for production without errors.

**Other relevant information:**

➜ ./node_modules/.bin/webpack --version

System:
  OS: macOS 11.4
  CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  Memory: 1.78 GB / 16.00 GB
Binaries:
  Node: 18.18.2 - ~/.nvm/versions/node/v18.18.2/bin/node
  Yarn: 1.22.21 - ~/.nvm/versions/node/v18.18.2/bin/yarn
  npm: 9.8.1 - ~/.nvm/versions/node/v18.18.2/bin/npm
Browsers:
  Brave Browser: 103.1.41.96
  Chrome: 119.0.6045.159
  Safari: 14.1.1
Packages:
  babel-loader: ^9.1.3 => 9.1.3 
  clean-webpack-plugin: ^4.0.0 => 4.0.0 
  css-loader: ^6.8.1 => 6.8.1 
  eslint-loader: ^1.7.1 => 1.9.0 
  eslint-webpack-plugin: ^4.0.1 => 4.0.1 
  sass-loader: ^13.3.2 => 13.3.2 
  style-loader: ^3.3.3 => 3.3.3 
  terser-webpack-plugin: ^5.3.9 => 5.3.9 
  uglifyjs-webpack-plugin: ^1.2.4 => 1.2.4 
  webpack: ^5.88.2 => 5.89.0 
  webpack-cli: ^5.1.4 => 5.1.4 
  webpack-dev-server: ^4.15.1 => 4.15.1 
  webpack-merge: ^5.10.0 => 5.10.0

**Comment by alexander-akait** [comment]
Please provide reproducible test repo, maybe you have broken JS syntax...

**Reply by nullstd** [reply]
This should be easy to reproduce with any repo, such as the one from https://redux.js.org/tutorials/essentials/part-1-overview-concepts, I found it happens when `TerserPlugin.uglifyJsMinify` is used and `keep_classnames` is `true`

**Reply by alexander-akait** [reply]
Can you create reproduciable test repo, because it works for me...

**Reply by varit05** [reply]
I faced this issue and came across this thread. 

I've created a test [repo](https://github.com/varit05/webpack-terser-webpack-plugin-with-uglify-js) where this is still producible. 

I have tried the steps from the webpack [documentation](https://webpack.js.org/plugins/terser-webpack-plugin/#uglify-js) 

Happy to help in any capacity.

**Reply by alexander-akait** [reply]
@varit05 I see, the problem is ``deadCode` is not a supported option` by uglify-js, WIP on fix to output better errors here

**Reply by alexander-akait** [reply]
@varit05 There is a fix - https://github.com/webpack/terser-webpack-plugin/pull/641, release will be soon

**Reply by varit05** [reply]
@alexander-akait Thanks for the quick resolution, I really appreciate it.

**Reply by varit05** [reply]
I’m happy to confirm that the plugin now provides a clearer error with the `terserOptions`. I have updated the [repo](https://github.com/varit05/webpack-terser-webpack-plugin-with-uglify-js) and played with the different `terserOptions`. 

Thank you @alexander-akait for your prompt response.
